src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedArreys: GroupsMap<T> = {};
```

Comment:
There is a typo in the variable name `groupedArreys`; it should be `groupedArrays` for clarity and correct spelling.